prow/cmd/generic-autobumper/main.go

```diff
@@ -476,14 +484,18 @@ func getVersionsAndCheckConsistency(prefixes []prefix, images map[string]string)
 	versions := map[string][]string{}
 	consistencyChecker := map[string]string{}
 	for _, prefix := range prefixes {
+		exceptions := sets.NewString(prefix.ConsistentImageExceptions...)
 		for k, v := range images {
 			if strings.HasPrefix(k, prefix.Prefix) {
+				image := imageFromName(k)
 				found, ok := consistencyChecker[prefix.Prefix]
-				if ok && (found != v) && prefix.ConsistentImages {
-					return nil, fmt.Errorf("%s:%s not bumped consistently for prefix %s(%s), expected version: %s", k, v, prefix.Prefix, prefix.Name, found)
-				} else if !ok {
+				// If consistency should be checked and this image is in ConsistentImageExceptions, record it.
+				if !ok && prefix.ConsistentImages && !exceptions.Has(image) {
 					consistencyChecker[prefix.Prefix] = v
 				}
+				if ok && (found != v) && prefix.ConsistentImages && !exceptions.Has(image) {
```

Comment:
Nit: I'm not sure why you flipped the order of conditions from before --- any particular reason?

But also, instead of duplicating the `prefix.ConsistentImages && !exceptions.Has(image)` here and earlier, I would just do it once and then have the other boolean checks inside it to keep things DRY.

Reply:
No particular one, I was being verbose while refactoring to make sure I wasn't missing cases. Updated with the suggestion and flipped the internals to do the return first.